frontend/src/components/forms/EditLog.tsx

```diff
@@ -0,0 +1,465 @@
+  };
+
+  const handleResidentChange = (
+    selectedOption: SingleValue<{ label: string; value: number }>,
+  ) => {
+    if (selectedOption !== null) {
+      setResident(selectedOption.value);
+      setResidentError(false);
+    }
+  };
+
+  const handleTagsChange = (
+    selectedTags: MultiValue<{ label: string; value: string }>,
+  ) => {
+    const newTagsList = selectedTags.map((tag) => tag.value);
+    setTags(newTagsList);
+  };
+
+  const handleAttnToChange = (
+    selectedOption: SingleValue<{ label: string; value: number }>,
+  ) => {
+    if (selectedOption !== null) {
+      setAttnTo(selectedOption.value);
+    } else {
+      setAttnTo(-1);
+    }
+  };
+
+  const handleNotesChange = (e: { target: { value: unknown } }) => {
+    const inputValue: string = e.target.value as string;
+    setNotes(inputValue);
+    setNotesError(inputValue === "");
+  };
+
+  const initializeValues = () => {
+    // reset state variables 
+    setEmployee(getCurUserSelectOption());
+    setDate(new Date(logRecord.datetime));
+    setTime(
+      date.toLocaleTimeString([], {
+        hour: "2-digit",
+        minute: "2-digit",
+        hour12: false,
+      }),
+    );
+    setBuilding(logRecord.building);
+    const residentId = residentOptions.find((item) => item.label === logRecord.residentId)?.value;
+    setResident(residentId !== undefined ? residentId : -1);
+    setTags(logRecord.tags);
+    setAttnTo(logRecord.attnTo !== undefined ? logRecord.attnTo : -1);
+    setNotes(logRecord.note);
+    setFlagged(logRecord.flagged);
+
+    // error states for non-nullable inputs
+    setEmployeeError(false);
+    setDateError(false);
+    setTimeError(false);
+    setBuildingError(false);
+    setResidentError(false);
+    setNotesError(false);
+  };
+
+  const handleSubmit = () => {
+    // Update error states
+    setEmployeeError(!employee.label);
+    setDateError(date === null);
+    setTimeError(time === "");
+    setBuildingError(building === "");
+    setResidentError(resident === -1);
+    setNotesError(notes === "");
+
+    // If any required fields are empty, prevent form submission
+    if (
+      !employee.label ||
+      date === null ||
+      time === "" ||
+      building === "" ||
+      resident === -1 ||
+      notes === ""
+    ) {
+      return;
+    }
+
+    LogRecordAPIClient.editLogRecord({
+      logId: logRecord.logId,
+      employeeId: employee.value,
+      residentId: resident,
+      datetime: combineDateTime(date, time),
+      flagged,
+      note: notes,
+      tags,
+      building,
+      attnTo: attnTo === -1 ? undefined : attnTo,
+    }).then((res) => {
+      if (res != null) {
+        setAlertData(ALERT_DATA.SUCCESS);
+
+        countRecords();
+        getRecords(1);
+        setUserPageNum(1);
+
+        toggleClose();
+      } else {
+        setAlertData(ALERT_DATA.ERROR);
+      }
+      setShowAlert(true);
+    });
+  };
+
+  useEffect(() => {
+    initializeValues();
```

Comment:
`if (isOpen) {
initializeValues();
}`

Just to ensure this is only called when the modal opens